.github/workflows/test.yml

```
@@ -0,0 +1,157 @@
+      - run: mvn -pl :zeebe-elasticsearch-exporter -B -D skipUTs -D skipChecks -Dfailsafe.rerunFailingTestsCount=3 verify
+      - name: Archive Test Results
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: Exporter test results
+          path: "**/target/failsafe-reports/"
+          retention-days: 7
+  project-list:
+    name: List projects
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: sudo apt-get -y install jq
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - name: Get Project List
+        id: list-projects
+        run: >
+          echo "::set-output name=fast::$(mvn -pl !:zeebe-workflow-engine,!:zeebe-logstreams -Dexec.executable='echo' -Dexec.args='${project.artifactId}' exec:exec -q |  jq -cnR [inputs])"
+    outputs:
+      fast: ${{ steps.list-projects.outputs.fast }}
+  unit-tests:
+    name: Unit tests
+    needs: project-list
+    strategy:
+      fail-fast: false
+      matrix:
+        project: ${{ fromJson(needs.project-list.outputs.fast) }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - run: mvn -T1C -B -D skipTests -D skipChecks install
+      - run: mvn -B -D skipITs -D skipChecks verify -pl :${{ matrix.project }}
+      - name: Archive Test Results
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: Unit test results for ${{ matrix.project }}
+          path: "**/target/surefire-reports/"
+          retention-days: 7
+  slow-unit-tests:
+    name: Slow unit tests
+    runs-on: ["n1-standard-8-netssd-preempt"]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - uses: stCarolas/setup-maven@v4.3
+        with:
+          maven-version: 3.8.5
+      - run: mvn -T1C -B -D skipTests -D skipChecks package
+      - run: mvn -B -D skipITs -D skipChecks -pl :zeebe-workflow-engine,:zeebe-logstreams verify
+      - name: Archive Test Results
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: Slow unit test results
+          path: "**/target/surefire-reports/"
+          retention-days: 7
+  smoke-tests:
+    # This name is hard-referenced from bors.toml
+    # Remember to update that if this name, or the matrix.os changes
+    name: Smoke tests on ${{ matrix.os }}
+    timeout-minutes: 20
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-latest, windows-2022, ubuntu-latest ]
```

```suggestion
        os: [ macos-latest, windows-latest, ubuntu-latest ]
```

🔧 We could change this to windows-latest since they recently changed latest to 2022